lib/ss_light.go

```diff
@@ -229,31 +237,35 @@ func NewLightClient(
 }
 
 type ProgressUpdater interface {
-	UpdateProgress(int)
+	UpdateProgress(int, int, int)
 }
 
 func (l *LightClient) Start(
 	sharable string,
 	onlyInfo bool,
 	stat bool,
 	progUpd ProgressUpdater,
-) (string, error) {
+) *Out {
 	metadata, err := getInfo(sharable, "", l.pubKey)
 	if err != nil {
 		log.Errorf("Failed getting metadata Err: %s", err.Error())
-		return "Failed getting metadata", err
+		return NewOut(serviceError, "Failed getting metadata", err.Error(), nil)
 	}
-	log.Infof("Got metadata info %+v", metadata)
 
+	// STEP : Got metadata
+	showStep(success, "Got metadata", l.jsonOut)
+
+	log.Infof("Got metadata info %+v", metadata)
 	if onlyInfo {
-		return fmt.Sprintf("%+v\n", metadata), nil
+		return NewOut(success, MetaInfo, "", metadata)
 	}
-
-	dst, err := os.Create(
-		combineArgs(fpSeparator, l.destination, metadata.Cookie.Filename))
+	if l.destination == "." {
+		l.destination = combineArgs(fpSeparator, l.destination, metadata.Cookie.Filename)
+	}
+	dst, err := os.Create(l.destination)
 	if err != nil {
 		log.Errorf("Failed creating dest file Err: %s", err.Error())
-		return "Failed creating destination file", err
+		return NewOut(internalError, "Failed creating destination file", err.Error(), nil)
```

Comment:
This is a different type of error. This could only happen if user gave destination path incorrect